packages/wallet/api/src/marshal-contexts.js

```diff
@@ -0,0 +1,257 @@
+// @ts-check
+import { makeScalarMap } from '@agoric/store';
+import { Far, makeMarshal, Remotable } from '@endo/marshal';
+import { HandledPromise } from '@endo/eventual-send'; // TODO: convince tsc this isn't needed
+
+const { details: X, quote: q } = assert;
+
+/**
+ * Make context for exporting wallet data where brands etc. can be recognized by boardId.
+ *
+ * Objects should be registered before serialization:
```

Comment:
"should" doesn't convey the stakes here. Is the following accurate?

By default, objects get serialized with a fresh slot. There are two cases where this is bad:
1. The object is closely held, so the default behavior would expose it 
2. The object is published, so the default behavior would create a new object identity that can't be matched to the public one

Therefore to prevent these problems you must:
1. Use `initPurseId` for purse objects
2. Use `initBoardId` / `ensureBoardId` for published board objects (e.g. brands and issuers)

Reply:
Close, but the default behavior would not expose it.

I think the description of this PR makes a pretty good docstring here.